src/Stack/Options.hs

```diff
@@ -861,7 +861,7 @@ ghcVariantParser hide =
         readGHCVariant
         (long "ghc-variant" <> metavar "VARIANT" <>
          help
-             "Specialized GHC variant, e.g. integersimple (implies --no-system-ghc)" <>
+             "Specialized GHC variant, e.g. integersimple" <>
```

Comment:
Both your proposals are good. FWIW I understand "implies" like you do, I had just misunderstood the code.